tests/app/forms/field_handlers/test_select_handler.py

```diff
@@ -54,3 +56,35 @@ class TestForm(Form):
     assert form.test_field.label.text == radio_json["label"]
     assert form.test_field.description == radio_json["guidance"]
     assert form.test_field.choices == expected_choices
+
+
+def test_get_field_with_bad_choices(value_source_resolver):
```

Comment:
Is this still needed given the if choices check has been removed?. This shouldn't happen?.

Reply:
No this is necessary for  an empty choices list.

Reply:
What does that mean?. We should have had tests for choices that don't match the available choices?. I ask because the test is basically using an invalid schema.

Reply:
Lewis mentioned the same yet the schema apparently validates in the existing test using it - I used the same schema. So is the existing test bad?

I figured the new code for `SelectFieldWithDetailAnswer.pre_validate` needs coverage but as it happens I ran the unittests without that test and it still says 100% coverage so I guess that's alright

Reply:
Unless of course the schema isn't validated in the test